catroid/src/main/java/org/catrobat/catroid/common/ThreadScheduler.java

```diff
@@ -60,7 +73,7 @@ private void startThreadsInStartQueue() {
 	private void runThreadsForOneTick(Array<Action> actions, float delta) {
 		for (int i = 0; i < actions.size; i++) {
 			Action action = actions.get(i);
-			if (action.act(delta)) {
+			if (state == RUNNING && action.act(delta)) {
```

Comment:
I haven't looked up how unit-testable the `ThreadScheduler` is. Maybe you can test this line here with a unit test?

Reply:
Great idea, should be doable 👍 
I'll look into it!

Reply:
Done